bus-mapping/src/circuit_input_builder/input_state_ref.rs

```diff
@@ -1559,96 +1622,514 @@ impl<'a> CircuitInputStateRef<'a> {
     }
 
     /// Generate copy steps for bytecode.
+    #[allow(clippy::too_many_arguments)]
     pub(crate) fn gen_copy_steps_for_bytecode(
         &mut self,
         exec_step: &mut ExecStep,
         bytecode: &Bytecode,
-        src_addr: u64,
-        dst_addr: u64,
-        src_addr_end: u64,
-        bytes_left: u64,
-    ) -> Result<Vec<(u8, bool)>, Error> {
-        let mut copy_steps = Vec::with_capacity(bytes_left as usize);
-        for idx in 0..bytes_left {
-            let addr = src_addr.checked_add(idx).unwrap_or(src_addr_end);
-            let step = if addr < src_addr_end {
-                let code = bytecode.code.get(addr as usize).unwrap();
-                (code.value, code.is_code)
+        src_addr: usize,
+        dst_addr: usize,
+        src_addr_end: usize,
+        bytes_left: usize,
+        memory_updated: Memory,
```

Comment:
why not memory_updated: &Memory,

Reply:
should be good idea, will try .

Reply:
updated in [980b747](https://github.com/scroll-tech/zkevm-circuits/pull/466/commits/980b747e888c4b731cf07270a40857b7ef521471) thanks !